cmd/minikube/cmd/status.go

```diff
@@ -203,59 +204,63 @@ var statusCmd = &cobra.Command{
-			}
-
-			st, err := nodeStatus(api, *cc, *n)
-			if err != nil {
-				klog.Errorf("status error: %v", err)
-			}
-			statuses = append(statuses, st)
-		} else {
-			for _, n := range cc.Nodes {
-				machineName := driver.MachineName(*cc, n)
-				klog.Infof("checking status of %s ...", machineName)
-				st, err := nodeStatus(api, *cc, n)
-				klog.Infof("%s status: %+v", machineName, st)
+			if nodeName != "" || statusFormat != defaultStatusFormat && len(cc.Nodes) > 1 {
+				n, _, err := node.Retrieve(*cc, nodeName)
+				if err != nil {
+					exit.Error(reason.GuestNodeRetrieve, "retrieving node", err)
+				}
 
+				st, err := nodeStatus(api, *cc, *n)
 				if err != nil {
 					klog.Errorf("status error: %v", err)
 				}
-				if st.Host == Nonexistent {
-					klog.Errorf("The %q host does not exist!", machineName)
-				}
 				statuses = append(statuses, st)
-			}
-		}
-
-		switch strings.ToLower(output) {
-		case "text":
-			for _, st := range statuses {
-				if err := statusText(st, os.Stdout); err != nil {
-					exit.Error(reason.InternalStatusText, "status text failure", err)
+			} else {
+				for _, n := range cc.Nodes {
+					machineName := driver.MachineName(*cc, n)
+					klog.Infof("checking status of %s ...", machineName)
+					st, err := nodeStatus(api, *cc, n)
+					klog.Infof("%s status: %+v", machineName, st)
+
+					if err != nil {
+						klog.Errorf("status error: %v", err)
+					}
+					if st.Host == Nonexistent {
+						klog.Errorf("The %q host does not exist!", machineName)
+					}
+					statuses = append(statuses, st)
 				}
 			}
-		case "json":
-			// Layout is currently only supported for JSON mode
-			if layout == "cluster" {
-				if err := clusterStatusJSON(statuses, os.Stdout); err != nil {
-					exit.Error(reason.InternalStatusJSON, "status json failure", err)
+
+			switch strings.ToLower(output) {
+			case "text":
+				for _, st := range statuses {
+					if err := statusText(st, os.Stdout); err != nil {
+						exit.Error(reason.InternalStatusText, "status text failure", err)
+					}
 				}
-			} else {
-				if err := statusJSON(statuses, os.Stdout); err != nil {
-					exit.Error(reason.InternalStatusJSON, "status json failure", err)
+			case "json":
+				// Layout is currently only supported for JSON mode
+				if layout == "cluster" {
+					if err := clusterStatusJSON(statuses, os.Stdout); err != nil {
+						exit.Error(reason.InternalStatusJSON, "status json failure", err)
+					}
+				} else {
+					if err := statusJSON(statuses, os.Stdout); err != nil {
+						exit.Error(reason.InternalStatusJSON, "status json failure", err)
+					}
 				}
+			default:
+				exit.Message(reason.Usage, fmt.Sprintf("invalid output format: %s. Valid values: 'text', 'json'", output))
 			}
-		default:
-			exit.Message(reason.Usage, fmt.Sprintf("invalid output format: %s. Valid values: 'text', 'json'", output))
+			if !watch {
+				os.Exit(exitCode(statuses))
+			}
+			time.Sleep(1 * time.Second)
```

Comment:
how about making this configureable ?
like --watch-interval with a default value of 1 second?

Reply:
like the idea!
and instead of adding a separate flag, would we want just to expand the `--watch` flag to also support an optional value (defaulting to 1 sec)? eg, `--watch 5` would sleep for 5 sec between consecutive polls